### PR TITLE
Feature/introduce new input support

### PR DESCRIPTION
**Pull request by tomwim**

### CODE REVIEW

File: Runtime/Spectator/SpectatorController.cs [hunk 1]
```diff
@@ -8,7 +11,7 @@ namespace Innoactive.Creator.UX
     /// </summary>
     public class SpectatorController : MonoBehaviour
     {
-        /// <summary>
+        /// <summary>spe
```
Comment (Gusinuhe) [inline Runtime/Spectator/SpectatorController.cs:14]:
```suggestion
        /// <summary>
```

🤔

File: Runtime/Spectator/SpectatorSettings.cs [hunk 2]
```diff
@@ -10,9 +15,61 @@ namespace Innoactive.Creator.UX
...
+            }
+
+            string assetName = originalAssetName.Split('.')[0];
+            InputActionAsset inputActionAsset = Resources.Load<InputActionAsset>(assetName);
+
+            string path = Path.Combine(Application.dataPath, targetPath);
+            string assetPath = Path.Combine("Assets", targetPath, targetAssetName);
+
+            Directory.CreateDirectory(path);
+            if (AssetDatabase.CopyAsset(AssetDatabase.GetAssetPath(inputActionAsset), assetPath))
```
Comment (Gusinuhe) [inline Runtime/Spectator/SpectatorSettings.cs:53]: These method might not be called on runtime, but `AssetDatabase` is part of `UnityEditor` and won't compile.

**Review by Gusinuhe — CHANGES_REQUESTED** [review]

File: Runtime/Spectator/SpectatorSettings.cs (hunk 2 above)
Comment (SimonTheSourcerer) [reply Runtime/Spectator/SpectatorSettings.cs:53]: Yes it does not build~

**Review by SimonTheSourcerer — CHANGES_REQUESTED** [review]

File: Runtime/Spectator/SpectatorController.cs [hunk 3]
```diff
@@ -23,12 +26,21 @@ protected virtual void Update()
         /// </summary>
         protected virtual void HandleInput()
         {
+#if !ENABLE_INPUT_MANAGER && ENABLE_LEGACY_INPUT_MANAGER
```
Comment (Gusinuhe) [inline Runtime/Spectator/SpectatorController.cs:29]:
```suggestion
#if !ENABLE_INPUT_SYSTEM && ENABLE_LEGACY_INPUT_MANAGER
```

**Review by Gusinuhe — CHANGES_REQUESTED** [review]
There is still a possibility for `SpectatorSettings.KeyBindingSettings` to be null.
Perhaps make `KeyBindingSettings` a property and load `Runtime/Spectator/Resources/SpectatorKeyBindings.inputactions` when null and is not editor 🤔

**Comment by tomwim** [comment]
Closed in favor of https://github.com/Innoactive/Basic-UI-Component/pull/29

**Pull request closed without merging**